pkg/cluster/inventory.go

```diff
@@ -54,28 +56,51 @@ func (i *DefaultInventory) WithTx(tx *db.TxConnection) (Inventory, error) {
 	return NewInventory(tx, i.Debug, i.metricsCollector)
 }
 
+// Used as tables for GORM Query
+type inventoryClusters struct{}
+type inventoryClusterConfigs struct{}
+type inventoryClusterConfigStatus struct{}
+
 func (i *DefaultInventory) CountRetries(runtimeID string, configVersion int64, maxRetries int, errorStatus ...model.Status) (int, error) {
 	if len(errorStatus) == 0 {
 		return 0, errors.New("errorStatus slice is empty")
 	}
 
 	var maxStatusHistoryLength = maxRetries * 5 //cluster can have three interims state between errors, thus 5 is more than enough
-	q, err := db.NewQuery(i.Conn, &model.ClusterStatusEntity{}, i.Logger)
+	q, err := db.NewQueryGorm(i.Conn, &model.ClusterStatusEntity{}, i.Logger)
+
 	if err != nil {
 		return 0, errors.Wrap(err, fmt.Sprintf("failed to initialize query for runtime %s", runtimeID))
 	}
-	clusterStatuses, err := q.Select().Where(map[string]interface{}{"RuntimeID": runtimeID, "ConfigVersion": configVersion}).OrderBy(map[string]string{"ID": "desc"}).Limit(maxStatusHistoryLength).GetMany()
+	clusterStatusesSQL := q.Query().Select("*").
+		Where("runtime_id = @runtime AND config_version = @configversion", sql.Named("runtime", runtimeID), sql.Named("configversion", configVersion)).
+		Order("id desc").
+		Limit(maxStatusHistoryLength).
+		Find(inventoryClusterConfigStatus{})
 	if err != nil {
 		return 0, errors.Wrap(err, fmt.Sprintf("failed to count error for runtime %s", runtimeID))
 	}
+	dataRows, err := i.Conn.QueryGorm(clusterStatusesSQL)
+	if err != nil {
+		return 0, err
+	}
 
 	errCnt := 0
-	for _, clusterStatus := range clusterStatuses {
-		clStatusEntity := clusterStatus.(*model.ClusterStatusEntity)
-		if clStatusEntity.Status.IsFinal() {
-			if statusInSlice(clStatusEntity.Status, errorStatus) {
+	for dataRows.Next() {
```

Comment:
In this PR I am using GORM just for the SQL building and I still use our Query methods from the `Connection` interface. Thus, the `Find()` method is actually not used to execute the query, as it would normally do. Here, the `Find` method only is used for to add the `FROM` clause in the SQL String.
This behaviour can be changed by setting the DryRun mode in the Gorm configuration, which I also did in this PR. Meaning the "result" of such line (i.e. [here](https://github.com/kyma-incubator/reconciler/pull/1029/files#diff-f921d302f6db810d79d7007b3c8a95dc107a6a7e4da95f92f3a96cbb6f6589aaR77)) is just the SQL with its vars in a gorm struct. Thus it still can be used as a sub-query in other queries, like [here](https://github.com/kyma-incubator/reconciler/pull/1029/files#diff-f921d302f6db810d79d7007b3c8a95dc107a6a7e4da95f92f3a96cbb6f6589aaR708-R713).

But what we could also do, is using the `Find` method to actually execute the queries and thus also using the built-in parser of gorm. But I did not had the time until know to test this feature in regards of compatibility with our `tx` object.

I would suggest to create a follow-up issue for that.

This, btw also resolves the other comment from you (https://github.com/kyma-incubator/reconciler/pull/1029#discussion_r849157163)